ssl/statem/statem_srvr.c

```diff
@@ -1318,7 +1318,7 @@ MSG_PROCESS_RETURN tls_process_client_hello(SSL *s, PACKET *pkt)
         }
 
         if (session_id_len > SSL_MAX_SSL_SESSION_ID_LENGTH) {
-            al = SSL_AD_DECODE_ERROR;
+            al = SSL_AD_ILLEGAL_PARAMETER;
```

Comment:
It's probably debatable. The way I saw it we have a syntactically correct ciphersuite_len, session_id_len and challenge_len, i.e. we have successfully parsed the data structure. However on inspection of the data we find a parameter that has a value that should not be allowed, so it's an illegal parameter.